aptos-move/aptos-gas-meter/src/meter.rs

```diff
@@ -489,7 +490,28 @@ where
     }
 
     fn charge_io_gas_for_write(&mut self, key: &StateKey, op: &WriteOp) -> VMResult<()> {
-        let cost = self.storage_gas_params().pricing.io_gas_per_write(key, op);
+        let cost = self
+            .storage_gas_params()
+            .pricing
+            .io_gas_per_write(key, op, None);
+
+        self.algebra
+            .charge_io(cost)
+            .map_err(|e| e.finish(Location::Undefined))
+    }
+
+    fn charge_io_gas_for_group_write(
+        &mut self,
+        key: &StateKey,
+        group_write: &GroupWrite,
+    ) -> VMResult<()> {
+        let group_size = group_write.encoded_group_size()?;
+        let cost = self.storage_gas_params().pricing.io_gas_per_write(
+            key,
+            &group_write.metadata_op,
+            Some(group_size),
```

Comment:
this is a weird modification to the api, I'd just have a new api to charge for the group write or have a trait to read the value size